src/data_types/mod.rs

```diff
@@ -1,5 +1,14 @@
-/// A pointer to an opaque data structure.
-pub type Handle = *mut ();
+use core::ffi::c_void;
+
+/// A collection of related interfaces
```

Comment:
This isn't 100% correct, UEFI handles are equivalent to WinAPI handles, i.e. they uniquely identify a UEFI image (app or driver). Handles are used in other places in the API for more than just querying protocols.

Reply:
I was just paraphrasing the UEFI spec here, since I did not understand why you deviated from it. Will think of a different wording which encompasses the full generality of the concept.